oauthproxy_test.go

```diff
@@ -493,6 +493,53 @@ func TestSignInPageIncludesTargetRedirect(t *testing.T) {
 	}
 }
 
+func TestGetRedirect(t *testing.T) {
+	testCases := []struct {
+		name       string
+		options    func() *Options
+		requestURI string
+		expected   string
+	}{
+		{
+			name: "Redirect to root",
+			options: func() *Options {
+				return testOptions()
+			},
+			requestURI: "/oauth2/start",
+			expected:   "/",
+		},
+		{
+			name: "Redirect to root",
+			options: func() *Options {
+				opts := testOptions()
+				opts.SkipProviderButton = true
+				opts.ProxyPrefix = "/oauth"
+				return opts
+			},
+			requestURI: "/oauth/start",
+			expected:   "/",
+		},
+		{
+			name: "Redirect to custom uri",
+			options: func() *Options {
+				opts := testOptions()
+				opts.SkipProviderButton = true
+				opts.ProxyPrefix = "/oauth2"
+				return opts
+			},
+			requestURI: "/oauth2/start",
+			expected:   "/oauth2/start",
+		},
```

Comment:
Why do these scenarios behave differently? Why is the `/oauth` special-cased in the code above?

Reply:
`/oauth` is default value for `ProxyPrefix`. the expected redirect is `/`. that is our case. I have described it in the description.

`/oauth2` is custom value for `ProxyPrefix`. Keep the behaviour as it is.

Reply:
I still don't see the reason why we should treat `/oauth` differently than any other prefix

Reply:
I do not know if my case is used by other customers. my case is using default setting with `skip-provider-button=true`. so it always redirects to `oauth/start`. it should be redirected to `/`. You can see there is not such fix introduced in #18 in upstream - https://github.com/bitly/oauth2_proxy/blob/fa2771998a98a5bfdfa3c3503757668ac4f1c8ec/oauthproxy.go#L422